apple-nav/src/App.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react';
+import './App.css';
+import { Route } from 'react-router-dom'
+import NavWrapper from './components/NavWrapper'
+import SubNav from './components/SubNav'
+import NewMac from './components/Displays/IndividualProducts/NewMac'
+import NewIpad from './components/Displays/IndividualProducts/NewIpad'
+import NewIphone from './components/Displays/IndividualProducts/NewIphone'
+import NewMusic from './components/Displays/IndividualProducts/NewMusic'
+import NewTv from './components/Displays/IndividualProducts/NewTv'
+import NewWatch from './components/Displays/IndividualProducts/NewWatch'
+import { macData, ipadData, iphoneData, watchData, tvData, musicData} from './AppleIconsData'
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      mac: [],
+      ipad: [],
+      iphone: [],  
+      watch: [],
+      tv: [],
+      music: [],
+    }
+  }
+
+  componentDidMount() {
+    this.setState ({mac: macData})
```

Comment:
You don't need to make this many setState calls. You can do all of this with one call:
```suggestion
    this.setState ({
    mac: macData,
    ipad: ipadData,
    iphone: iphoneData,
    watch: watchData,
    tv: tvData,
    music: musicData
    })
```